apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.Tests/BigQueryInsertResultsTests.cs

```diff
@@ -0,0 +1,426 @@
+            Assert.Empty(results.Errors);
+        }
+
+        [Fact]
+        public void SomeRowsInserted()
+        {
+            ErrorProto row1Error1 = new ErrorProto { Location = "field_1", Reason = "reason_1", Message = "message_1" };
+            ErrorProto row1Error2 = new ErrorProto { Location = "field_2", Message = "message_2" };
+            InsertErrorsData row1List1 = new InsertErrorsData { Index = 1, Errors = new List<ErrorProto> { row1Error1, row1Error2 } };
+
+            ErrorProto row5Error1 = new ErrorProto { Location = "field_3", Reason = "reason_3" };
+            InsertErrorsData row5List1 = new InsertErrorsData { Index = 5, Errors = new List<ErrorProto> { row5Error1 } };
+
+            InsertErrorsData row6List1 = new InsertErrorsData { Index = 6 };
+
+            ErrorProto row1Error3 = new ErrorProto { Location = "field_4", Reason = "reason_4", Message = "message_4" };
+            InsertErrorsData row1List2 = new InsertErrorsData { Index = 1, Errors = new List<ErrorProto> { row1Error3 } };
+
+            TableDataInsertAllResponse response = new TableDataInsertAllResponse
+            {
+                InsertErrors = new List<InsertErrorsData>
+                {
+                    row1List1,
+                    row5List1,
+                    row6List1,
+                    row1List2
+                }
+            };
+
+            IList<BigQueryInsertRow> rows = new List<BigQueryInsertRow>
+            { 
+                // 8 rows
+                new BigQueryInsertRow(),
+                new BigQueryInsertRow(),
+                new BigQueryInsertRow(),
+                new BigQueryInsertRow(),
+                new BigQueryInsertRow(),
+                new BigQueryInsertRow(),
+                new BigQueryInsertRow(),
+                new BigQueryInsertRow(),
+            };
+
+            List<BigQueryInsertRowErrors> expectedInsertRowErrors = new List<BigQueryInsertRowErrors>
+            {
+                new BigQueryInsertRowErrors(rows[1], new List<InsertErrorsData> { row1List1, row1List2 }),
+                new BigQueryInsertRowErrors(rows[5], new List<InsertErrorsData> { row5List1 }),
+                new BigQueryInsertRowErrors(rows[6], new List<InsertErrorsData> { row6List1 }),
+            };
+            List<SingleError> expectedSingleErrors = (from rowError in expectedInsertRowErrors
+                                                      from detail in rowError
+                                                      select detail).ToList();
+
+            BigQueryInsertResults results = new BigQueryInsertResults(
+                _client,
+                new InsertOptions { SkipInvalidRows = true },
+                rows,
+                response);
+
+            Assert.Equal(BigQueryInsertStatus.SomeRowsInserted, results.Status);
+            Assert.True(results.HasErrors);
+            Assert.Same(results, results.ThrowOnNoneInserted());
+            AssertException(
```

Comment:
Perhaps the Assert.Throws call could be in AssertException? Then it could use a method group conversion and read really nicely:

```csharp
AssertException(results.ThrowOnNoneInserted);
AssertException(results.ThrowOnNotAllInserted);
AssertException(results.ThrowOnAnyError);
```